src/Extensions/Hosting.Services/OmexStatelessService.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using System.Fabric;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.ServiceFabric.Services.Communication.Runtime;
+using Microsoft.ServiceFabric.Services.Runtime;
+
+namespace Microsoft.Omex.Extensions.Hosting.Services
+{
+	/// <summary>
+	/// Omex implementation of stateless service fabric service
+	/// </summary>
+	public sealed class OmexStatelessService : StatelessService, IServiceFabricService<StatelessServiceContext>
+	{
+		private readonly OmexStatelessServiceRunner m_serviceParameters;
+
+		internal OmexStatelessService(
+			OmexStatelessServiceRunner serviceRunner,
+			StatelessServiceContext serviceContext)
+				: base(serviceContext) => m_serviceParameters = serviceRunner;
+
+		/// <inheritdoc />
+		protected override IEnumerable<ServiceInstanceListener> CreateServiceInstanceListeners() =>
+			m_serviceParameters.ListenerBuilders.Select(b => new ServiceInstanceListener(c => b.Build(this), b.Name));
+
+		/// <inheritdoc />
+		protected override Task RunAsync(CancellationToken cancellationToken) =>
+			Task.WhenAll(m_serviceParameters.ServiceActions.Select(r => r.RunAsync(this, cancellationToken)));
```

Comment:
Does it make sens to .ConfigureAwait(false) here and in the stateful file?

Reply:
We are not awaiting tasks here, and if we call `.ConfigureAwait(false)` it will return Awaitable something so we won't be able to pass it as a parameter instead of task

Reply:
Ah right.